src/pages/api/contact-details/[id]/index.test.js

```diff
@@ -0,0 +1,171 @@
+
+    axios.mockImplementationOnce(() =>
+      Promise.resolve({
+        status: 200,
+        data: CONTACT_DETAILS,
+      })
+    )
+  })
+
+  describe('when the cookie is for an agent', () => {
+    const signedCookie = jsonwebtoken.sign(
+      {
+        name: 'name',
+        email: 'name@example.com',
+        groups: [AGENTS_GOOGLE_GROUPNAME],
+      },
+      HACKNEY_JWT_SECRET
+    )
+
+    const headers = {
+      'Content-Type': 'application/json',
+      'x-api-key': REPAIRS_SERVICE_API_KEY,
+      'x-hackney-user': signedCookie,
+      Authorization: signedCookie,
+    }
+
+    test('the response can include full contact information', async () => {
+      const req = createRequest({
+        method: 'get',
+        headers: { Cookie: `${GSSO_TOKEN_NAME}=${signedCookie};` },
+        query: {
+          id: TENURE_ID,
+        },
+        params: {},
+      })
+
+      const res = createResponse()
+
+      await contactDetailsEndpoint(req, res)
+
+      expect(axios).toHaveBeenCalledTimes(1)
+      expect(axios).toHaveBeenCalledWith({
+        method: 'get',
+        headers,
+        url: `${REPAIRS_SERVICE_API_URL}/contact-details/${TENURE_ID}`,
+        params: {},
+        paramsSerializer,
+        data: {},
+      })
+
+      const parsedData = JSON.parse(res._getData())
+
+      expect(parsedData).toEqual(CONTACT_DETAILS)
+    })
+  })
+
+  describe('when the cookie is for an authorisation manager', () => {
+    const signedCookie = jsonwebtoken.sign(
+      {
+        name: 'name',
+        email: 'name@example.com',
+        groups: [AUTHORISATION_MANAGERS_GOOGLE_GROUPNAME],
+      },
+      HACKNEY_JWT_SECRET
+    )
+
+    const headers = {
+      'Content-Type': 'application/json',
+      'x-api-key': REPAIRS_SERVICE_API_KEY,
+      'x-hackney-user': signedCookie,
+      Authorization: signedCookie,
+    }
+
+    test('the response can include full contact information', async () => {
+      const req = createRequest({
+        method: 'get',
+        headers: { Cookie: `${GSSO_TOKEN_NAME}=${signedCookie};` },
+        query: {
+          id: TENURE_ID,
+        },
+        params: {},
+      })
+
+      const res = createResponse()
+
+      await contactDetailsEndpoint(req, res)
+
+      expect(axios).toHaveBeenCalledTimes(1)
+      expect(axios).toHaveBeenCalledWith({
+        method: 'get',
+        headers,
+        url: `${REPAIRS_SERVICE_API_URL}/contact-details/${TENURE_ID}`,
+        params: {},
+        paramsSerializer,
+        data: {},
+      })
+
+      const parsedData = JSON.parse(res._getData())
+
+      expect(parsedData).toEqual(CONTACT_DETAILS)
+    })
+  })
+
+  describe('when the cookie is for a contractor', () => {
+    const signedCookie = jsonwebtoken.sign(
+      {
+        name: 'name',
+        email: 'name@example.com',
+        groups: [CONTRACTORS_GOOGLE_GROUPNAME],
+      },
+      HACKNEY_JWT_SECRET
+    )
+
+    test('response contact information is redacted', async () => {
```

Comment:
Is this testing the contact details endpoint, or the auth mechanisms & security?